auto_nag/scripts/topcrash_add_keyword.py

```diff
@@ -0,0 +1,165 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+import re
+
+from libmozdata.bugzilla import Bugzilla
+from libmozdata.connection import Connection
+
+from auto_nag import utils
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.topcrash import Topcrash
+
+# TODO: should be moved when resolving https://github.com/mozilla/relman-auto-nag/issues/1384
+HIGH_SEVERITY = {"S1", "critical", "S2", "major"}
+
+MAX_SIGNATURES_PER_QUERY = 30
+
+
+class TopcrashAddKeyword(BzCleaner):
+    def description(self):
+        return "Bugs with missing topcrash keywords"
+
+    def columns(self):
+        return ["id", "summary", "severity", "added_keyword"]
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+        if bugid in data:
+            return
+
+        has_topcrash_keyword = "topcrash" in bug["keywords"]
+        has_startup_keyword = "topcrash-startup" in bug["keywords"]
+        if has_topcrash_keyword and has_startup_keyword:
+            return
+
+        signatures = utils.get_signatures(bug["cf_crash_signature"])
+        has_topcrash_signature = any(
+            signature in self.topcrashes for signature in signatures
+        )
+        assert has_topcrash_signature
+
+        has_startup_signature = any(
+            self.topcrashes.get(signature) for signature in signatures
+        )
+
+        should_add_keyword = (
+            has_topcrash_signature
+            and not (has_topcrash_keyword or has_startup_signature)
+        ) or (has_startup_signature and not has_startup_keyword)
+
+        if not should_add_keyword:
+            return
+
+        keyword_to_add = "topcrash-startup" if has_startup_signature else "topcrash"
+
+        autofix = {
+            "keywords": {
+                "add": keyword_to_add,
+            },
+            "comment": {
+                "body": "The bug is linked to a startup topcrash signature."
+                if has_startup_signature
+                else "The bug is linked to a topcrash signature.",
```

Comment:
Maybe we should mention the rank or both the rank and the volume

Reply:
The rank and the volume can differ based on the query. The same signature will have different numbers when we query different channels, startup/not startup, or different platforms (https://github.com/mozilla/relman-auto-nag/pull/1451#discussion_r885013076). At the same time, this information is already available on Bugzilla using the extension that @calixteman built.

In general, I agree with the idea of giving more insights about why we adding this tag. Links to the signatures that drove us to add the keyword could be helpful (on crash-stats.mozilla.org).

Reply:
Now after 87cf2f2224edd87e96f9244a7de0e52a2f779e70, we have a well defined criteria to add the keywords. We could use these criteria to explain why we are adding a `topcrash` keyword. This will give more transparency and help identify if something is going wrong with the bot.

**_Example:_**
```
The bug is linked to top-crash signatures that are ranked in the top 20 desktop 
browser crashes on Release and the top 10 desktop browser crashes on Nightly.
```

**_Or:_**
```
The bug is linked to top-crash signatures that are ranked in:
- Top 20 desktop browser crashes on Release.
- Top 10 desktop browser crashes on Nightly.
```

Reply:
Done in bbb26f8c57dd69f4e79c5be2741a40e5f8fde7e3

## Example of a comment:

The bug is linked to a topcrash signature, which matches the following [criteria](https://wiki.mozilla.org/CrashKill/Topcrash):
- Top 20 desktop browser crashes on Release (startup)
- Top 10 content process crashes on Beta and Release

:sdetar, could you consider increasing the severity of this top-crash bug?

For more information, please visit [auto_nag documentation](https://wiki.mozilla.org/Release_Management/autonag#topcrash_add_keyword.py).